search/django/weedid/views.py

```diff
@@ -75,8 +79,80 @@ def upload(request):
-        weedcoco_path = store_tmp_weedcoco(file_weedcoco, upload_dir)
-        create_upload_entity(weedcoco_path, upload_id, user.id)
+        store_tmp_weedcoco(file_weedcoco, upload_dir)
+        create_upload_entity(upload_id, user.id)
+    except JsonValidationError as e:
+        traceback.print_exc()
+        return HttpResponseBadRequest(json.dumps(e.get_error_details()))
+    except Exception as e:
+        traceback.print_exc()
+        return HttpResponseBadRequest(str(e))
+    else:
+        return HttpResponse(
+            json.dumps(
+                {"upload_id": upload_id, "images": images, "categories": categories}
+            )
+        )
+
+
+def upload_voc(request):
+    if not request.method == "POST":
+        return HttpResponseNotAllowed(request.method)
+    user = request.user
+    if not (user and user.is_authenticated):
+        return HttpResponseForbidden("You dont have access to proceed")
+    try:
+        voc_dir = request.POST["voc_dir"]
+        voc = request.FILES["voc"]
+        upload_dir = os.path.join(UPLOAD_DIR, str(user.id), voc_dir)
+        store_tmp_voc(voc, upload_dir)
+    except Exception as e:
+        traceback.print_exc()
+        return HttpResponseBadRequest(str(e))
+    else:
+        return HttpResponse(f"Uploaded {voc.name} to {upload_dir}")
+
+
+def remove_voc(request):
+    if not request.method == "POST":
+        return HttpResponseNotAllowed(request.method)
+    user = request.user
+    if not (user and user.is_authenticated):
+        return HttpResponseForbidden("You dont have access to proceed")
+    voc_dir = request.POST["voc_dir"]
+    voc_name = request.POST["voc_name"]
+    voc_to_remove = os.path.join(UPLOAD_DIR, str(user.id), voc_dir, voc_name)
+    try:
+        if os.path.exists(voc_to_remove):
+            os.remove(voc_to_remove)
+            return HttpResponse(f"Removed {voc_name}")
+        else:
+            return HttpResponse(f"{voc_name} doesn't exist")
+    except Exception:
+        return HttpResponseBadRequest(f"Error when removing {voc_name}")
+
+
+def submit_voc(request):
+    if not request.method == "POST":
+        return HttpResponseNotAllowed(request.method)
+    user = request.user
+    if not (user and user.is_authenticated):
+        return HttpResponseForbidden("You dont have access to proceed")
+    voc_dir = request.POST["voc_dir"]
+    try:
+        images = []
+        weedcoco_json = voc_to_coco(
+            Path(os.path.join(UPLOAD_DIR, str(user.id), voc_dir))
+        )
+        validate(weedcoco_json, schema="coco")
+        for image_reference in weedcoco_json["images"]:
+            images.append(image_reference["file_name"].split("/")[-1])
+        categories = [
+            parse_category_name(category) for category in weedcoco_json["categories"]
+        ]
+        upload_dir, upload_id = setup_upload_dir(os.path.join(UPLOAD_DIR, str(user.id)))
+        store_tmp_voc_coco(weedcoco_json, upload_dir)
```

Comment:
We should either move or delete the VOC files.

Reply:
I set moving voc to upload_id/voc after user clicking the NEXT button, as after clicking submit button, user may still want to remove or modify voc. If the moving or deleting happen there, it will cause an error saying voc doesn't exist.